xarray/core/weighted.py

```diff
@@ -100,11 +103,22 @@ def __init__(self, obj, weights):
         if not isinstance(weights, DataArray):
             raise ValueError("`weights` must be a DataArray")
 
-        if weights.isnull().any():
-            raise ValueError(
-                "`weights` cannot contain missing values. "
-                "Missing values can be replaced by `weights.fillna(0)`."
+        def _weight_check(w):
+            if np.isnan(w).any():
```

Comment:
`.isnull()` does a bit more than that: `np.isnan` won't detect `NaT`. @mathause, how likely is it to get datetime-like arrays here? They don't make much sense as weights, but as far as I can tell we don't check (I might be missing something, though)

Reply:
There is no check for that. A `TimeDelta` may make some sense as weights. `DateTime` not so much. I think we can get away with using `np.isnan`. A `Date*` array as weights containing `NaT` should be super uncommon.

Reply:
We could still operate on the dataarray instead of the dask/numpy array, but as @dcherian suggesred, that would be less efficient. I would be curious as to what penalties would actually occur when we use the `weights.map_blocks` compared to `dask.array.map_blocks`?

Reply:
1. Could use `duck_array_ops.isnull` to account for `timedelta64`? It is weird to have it as a weight though. Does that work?
2. Re map_blocks: the xarray version adds tasks that create xarray objects wrapping every block in a dask array. That adds overhead which is totally unneccesary here.